### DISCUSSION TITLE
feature: Automatic map name change

### DISCUSSION THREAD

**Discussion by pandinocoder** [AscensionGameDev/Intersect-Engine#2713] [Ideas]
Currently, all maps need to be named manually. It's simple, but during the initial game development process, everything is named temporarily. Additionally, a feature that would allow naming and numbering a group of maps in a folder while using the map editor would be very helpful.

- [ ] Server configuration called `MapNameBehavior`
    - `FolderName` [Default] - this will set the default name to name of the folder the map is in, if it isn't in a folder it will fall back to the localized `New Map` string (`New Map` needs to be removed from the constructor, default name should be null)
    - `NumberedFolderName` - behaves the same as `FolderName`, except it will add a number to the end based on where the map is in the list (the `Name` property should still be `null`, this is only filled in at runtime)
    - `NewMap` - this is the existing behavior
- [ ] Developers should be able to specify different names for players and developers (add a `DeveloperName` property which is never shown to a player, even if `Name` is empty)
- [ ] Any maps with a value already stored in the `Name` property will be kept, this should only impact the name of new maps
- [ ] If a map has their `Name` set and then the name is cleared, it should set to `null` in the database, and since it is `null` at runtime it will fall back to another name according to the server configuration

Stretch goal, may not be feasible with the tree control in WinForms:
- [ ] Map folders should also have a `Name` and `DeveloperName`, the latter of which is exclusively shown to developers and never to players

**Comment by PyroTech03** [comment]
In the process, would love to see a "public name" along with a "dev name". Public name would be visible to the client and can share a name based on the region or whatever, while the dev name would allow us to have descriptive names that allows us to differentiate.

Ex. The dark forest, being a 4x4 grid. Could all be named "The Dark Forest" while dev names would be A1, A2.....D3, D4.
